pyposeidon/utils/cast.py

```diff
@@ -430,20 +434,32 @@ def run(self, **kwargs):
             logger.warning("meteo files present\n")
 
         # modify param file
-        rnday_new = (self.sdate - self.rdate).total_seconds() / (3600 * 24.0) + pd.to_timedelta(
-            self.time_frame
-        ).total_seconds() / (3600 * 24.0)
-        hotout_write = int(rnday_new * 24 * 3600 / info["params"]["core"]["dt"])
-        info["parameters"].update(
-            {
-                "ihot": 2,
-                "rnday": rnday_new,
-                "start_hour": self.rdate.hour,
-                "start_day": self.rdate.day,
-                "start_month": self.rdate.month,
-                "start_year": self.rdate.year,
-            }
-        )
+        if ihot == 2:
+            rnday_new = (self.sdate - self.rdate).total_seconds() / (3600 * 24.0) + pd.to_timedelta(
+                self.time_frame
+            ).total_seconds() / (3600 * 24.0)
```

Comment:
The formatting is ugly here. Either introduce new variables or maybe try some variation of this (I didn't test it):
``` python
            rnday_new = (self.sdate - self.rdate) + pd.to_timedelta(self.time_frame)
            rnday_new = rnday_new.total_seconds() / (3600 * 24.0)
            ...
```